### PR TITLE
feat: Only approve if a requested reviewer has approved.

### PR DESCRIPTION
**Pull request by chirino**
Not sure how to test this. @jimmidyson hows this look?  Got any testing tips?

### CODE REVIEW

**Comment by zregvart** [comment]
@syndesisio/all should we discuss this? This change would merge the PR as soon as all checks pass if no review was requested. Akka `commit-and-review` instead of `review-and-commit`.

**Comment by gashcrumb** [comment]
Didn't we have a discussion on this previously?  I seem to recall the team deciding on making reviews optional unless requested.  In which case this would be good, as then PRs don't stack up waiting to merge if the checks pass.

**Comment by rhuss** [comment]
So you have to request a review from a specific reviewer, right ?

There is no way to say 'i'd like to have a review but don't mind who ?' because then the PR would also get merged automatically.

From my feeling, the current workflow is not so bad because PR queue is not really too big. 
Also, as we are entering a more 'quality focussed' phase instead of the 'get features down' phase, maybe a quick look at the bug fix PRs make still sense ?

Actually, I could live with both

**Comment by gashcrumb** [comment]
Yeah, ideally you could say request a review from @syndesisio/backend for example or something but the little review request widget doesn't seem to support that :-)

**Review by zregvart — APPROVED** [review]
Seems that we're in agreement here to merge this, so let's do that, thanks @chirino :1st_place_medal:

**Comment by rhuss** [comment]
@zregvart ehm, wasn't the initial question how to test this ? So I assume that its currently untested. Let's keep fingers crossed. However I can jump on to it when I have some spare cycles, I darkly remember the setup ;-) ....

**Comment by zregvart** [comment]
@rhuss I had to revert it just now, seems that the WIP handling was broken somehow.

**Comment by zregvart** [comment]
> @rhuss I had to revert it just now, seems that the WIP handling was broken somehow.

By reverting I mean rolled back to previous deployment of the bot.

**Comment by zregvart** [comment]
I just noticed that the staging (hosted on the same cluster) is not working. I'm getting "Application is not available" even though the pods are all running. So that could be the reason it failed previously.

**Comment by rhuss** [comment]
ok, let me give a try ...

File: pkg/webhook/utils.go [hunk 1]
```diff
@@ -64,6 +64,58 @@ func labelsContainsLabel(labels []*github.Label, label string) bool {
 	return false
 }
 
+func doesPRNeedReview(pr *github.PullRequest, repo *github.Repository, gh *github.Client) (bool, error) {
+	reviewers, _, err := gh.PullRequests.ListReviewers(
```
Comment (rhuss) [inline pkg/webhook/utils.go:68]: @chirino  This request only returns the reviewers which do not have yet approved (just tested it). So we can't use this method to find out whether an initially requested reviewer has approved as it doesnt show up here anymore.

We need to find another way to tackle this (e.g by finding a proper API call to get the initial list of requested reviewers or to store the reviewer ids as part of an HTML comment in the PR description (a bit evolved though).

However, thats the reason why the status checks always fails even when an initially requested reviewer has approved // cc. @gashcrumb 

A first solution would be to switch off this check again.

Comment (rhuss) [reply pkg/webhook/utils.go:68]: Let me implement it like this: 

The "WIP" check (would probably rename it)
  * ... would be red if it's a WIP PR (titled or labeled as WIP)
  * ... would be red if the "status/review-requested" is given but no reviewer has approved
  * ... would be green if the label "status/review-requested" is not given
  * ... would be green if  the label "status/review-requested" is given and at least one reviewer (regardless who) has approved.
  
@deeleman @gashcrumb @chirino @zregvart would this fit our workflow ?

Comment (gashcrumb) [reply pkg/webhook/utils.go:68]: who applies the `status/review-requested` label?  The bot or the person opening the PR?

Other than that question I say ship it :-)

Comment (rhuss) [reply pkg/webhook/utils.go:68]: Its done by the bot automatically, as soon as a reviewer is requested. That way we can track also in retrospective the PRs for which reviewers were requested and which are unreviewed.

Comment (chirino) [reply pkg/webhook/utils.go:68]: @rhuss that sounds good to me.

**Pull request merged**